backend/object_detection.py

```
@@ -922,14 +974,21 @@ def enrich_detected_object(
     person_meta: List[Dict[str, Any]],
     attr_encoder: Any
 ) -> Dict[str, Any]:
-    """Helper to unify bounding-box extraction logic in both live and file stream."""
+    """
+    Helper to unify bounding-box extraction logic in both live and file stream.
+    
+    CRITICAL: Uses original unmasked ROI for color detection to avoid contamination
+    from artificial black background pixels in masked_roi.
+    """
     from pymongo import UpdateOne
```

Ruff (0.15.2)

[error] Redefinition of unused `UpdateOne` from line 26: `UpdateOne` redefined here

Remove definition: `UpdateOne`

(F811)